hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java

```diff
@@ -114,13 +118,11 @@ public void handle(@NonNull final HandleContext handleContext) {
         recordBuilder.nodeID(node.nodeId());
     }
 
-    private long getNextNodeID(WritableNodeStore nodeStore) {
+    private long getNextNodeID(@NonNull final WritableNodeStore nodeStore) {
+        requireNonNull(nodeStore);
         final var nodeIds = nodeStore.keys();
-        long max = 0;
-        while (nodeIds.hasNext()) {
-            max = Math.max(max, nodeIds.next().number());
-        }
-
-        return max + 1;
+        AtomicLong max = new AtomicLong(-1);
```

Comment:
This should be called only in single thread right ? Why do we need `AtomicLong`?

Reply:
Doesn't this work ?
```
// Convert the iterator to a stream
        Stream<NodeId> nodeIdStream = StreamSupport.stream(
            ((Iterable<NodeId>) () -> nodeIds).spliterator(), false
        );

        // Find the maximum node ID using the stream
        long maxNodeId = nodeIdStream
                .mapToLong(NodeId::number)
                .max()
                .orElse(-1); // Return -1 if the stream is empty

        // If the maximum node ID is less than 0, return -1
        return maxNodeId < 0 ? -1 : maxNodeId;
```

Reply:
IMO: nodeIds.forEachRemaining(nodeId -> max.set(Math.max(max.get(), nodeId.number()))); to much less code and concise. Long as nodeId is long.